src/vs/platform/terminal/common/capabilities/commandDetectionCapability.ts

```diff
@@ -79,6 +81,13 @@ export class CommandDetectionCapability implements ICommandDetectionCapability {
 			rows: this._terminal.rows
 		};
 		this._terminal.onResize(e => this._handleResize(e));
+		this._terminal.parser.registerCsiHandler({ final: 'J' }, params => {
+			if (params.length >= 1 && (params[0] === 2 || params[0] === 3)) {
+				this._clearCommandsInViewport();
+			}
+			// We don't want to override xterm.js' default behavior, just augment it
+			return false;
+		});
```

Comment:
This will need some verification on Windows before it gets merged to make sure. When conpty reprints the screen if it does one of these erase in displays it would break all the commands every now and then.

Reply:
Yes I'll wait until I have my windows computer to test before merging

Reply:
There's for sure some weirdness with that - esp when I resize